### PR TITLE
Fixing missing validation and adding shellcheck for submission scripts

### PR DESCRIPTION
**Pull request by ghost**
This PR moves out the submission scripts for better testing, adds shellcheck and fixes a few bugs.

This PR still needs testing.

### CODE REVIEW

File: .github/scripts/submit-provider-key.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+providername=$(echo "${BODY}" | grep "### Provider Name" -A2 | tail -n1 | tr "[:upper:]" "[:lower:]" | sed -e 's/[\r\n]//g')
+keydata=$(echo "${BODY}" | grep -A 1000 "BEGIN PGP PUBLIC KEY BLOCK"  | grep -B 1000 "END PGP PUBLIC KEY BLOCK")
+echo "${keydata}" > tmp.key
+
+if [[ ! "${namespace}" =~ ^[a-zA-Z0-9-]+$ ]]; then
+  gh issue comment "${NUMBER}" -b "Failed validation: Invalid namespace: '${namespace}'"
+  exit 1
+fi
+
+if [[ "${providername}" = "*No response*" ]]; then
```
Comment (Yantrio) [inline .github/scripts/submit-provider-key.sh:32]: I love it!

**Review by Yantrio — APPROVED** [review]

File: .github/scripts/submit-provider-key.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  exit 1
+fi
+
+if [[ -z "${providername}" ]]; then
+  keyfile="../keys/${namespace:0:1}/${namespace}/provider-$(date +%s).asc"
+else
+  keyfile="../keys/${namespace:0:1}/${namespace}/${providername}/provider-$(date +%s).asc"
+fi
+if [[ -d "$(dirname "${keyfile}")" ]]; then
+  msg=Updated
```
Comment (tom-reinders) [inline .github/scripts/submit-provider-key.sh]:
```suggestion
  msg="Updated"
```
If i'm being really syntax strict, not really a problem but nice to do

File: .github/scripts/submit-provider-key.sh [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  #git rm $(dirname $keyfile)/*
+else
+  msg=Created
```
Comment (tom-reinders) [inline .github/scripts/submit-provider-key.sh]:
```suggestion
  msg="Created"
```
Same as above

File: .github/workflows/shellcheck.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+name: Issue Submission to Pull Request
+on:
+  issues:
```
Comment (tom-reinders) [inline .github/workflows/shellcheck.yml:3]: Should this not be a check on PR's and/or pushes where any `./.github/scripts/*.sh` files are edited?

Comment (ghost) [reply .github/workflows/shellcheck.yml:3]: That makes it difficult to make the check required as it wouldn't always run.

**Pull request merged**